library/std/src/io/error.rs

```diff
@@ -341,6 +341,10 @@ pub enum ErrorKind {
     #[stable(feature = "unsupported_error", since = "1.53.0")]
     Unsupported,
 
+    /// The operation was partially sucessful and needs to be checked
+    /// later on due to not blocking
+    InProgress,
```

Comment:
You need an unstable annotation. Something like `#[unstable(feature = "io_error_inprogress", issue = "none")]`